ember-amount-input/.eslintrc.cjs

```diff
@@ -42,5 +42,83 @@ module.exports = {
       plugins: ['node'],
       extends: ['plugin:node/recommended'],
     },
+    // ts files
+    {
+      files: ['**/*.ts'],
+      extends: [
+        'plugin:@typescript-eslint/eslint-recommended',
+        'plugin:@typescript-eslint/recommended',
+      ],
+      rules: {
+        // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts
+        '@typescript-eslint/no-explicit-any': 'error',
+        '@typescript-eslint/no-non-null-assertion': 'error',
+        '@typescript-eslint/no-unused-vars': 'error',
+        // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/strict.ts
+        '@typescript-eslint/array-type': [
+          'error',
+          {
+            default: 'generic',
```

Comment:
There shouldn't be any (semantic) difference as `array` would just be the shorthand syntax for `generic`. Though I'm starting to wonder whether we might have problems with [`tuple`](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types) if we go with `generic` (I think we wouldn't be able to do type like: `[string, number]`)

Reply:
Yeah, good point, I think it's worth investigating. Especially if we're going to expand this same `.eslintrc` config to other repositories in the future 👍 

On my side, I think that `T[]` reads better because it's shorter and feels more natural to me, so I'd use `array`. What do you think?

Reply:
I tend to use the shorthand syntax a lot more, so I agree. I will update the `.eslintrc` config

Reply:
cool, I approved this in the meantime 🙂